### PR TITLE
[FIX] remove resource leaves when resetting to draft

### LINKED ISSUE #83: Löschen von Urlaub: Ressourcen werden nicht freigegeben
Wenn ein Mitarbeiter einen zukünftigen Urlaub löscht, wird ursprüngliche Zeitraum nicht freigegeben. Eine erneute Urlaubs-Beantragung scheitert.

Hintergrund:

Es wird zwar der Eintrag in `hr_leave` gelöscht. In `resource_calendar_leaves` bleibt die Abwesenheit bestehen und lediglich die Referenz auf `hr_leave` wird auf `NULL` gesetzt.

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

**Review by albig — CHANGES_REQUESTED** [review]
Funktioniert leider nicht. Jetzt kann der zukünftige Urlaub nicht mehr "abgelehnt" und "als Entwurf markiert" werden.

Fehlt da wieder ein `sudo()`?

**Comment by hbrunn** [comment]
ja, danke

**Comment by albig** [comment]
Das geht leider trotzdem nicht. Man muss ja erst auf "Auf Entwurf setzen" klicken und erhält dann folgendes:



Auf "Ablehnen" kann man ja als normaler Mitarbeiter ohnehin nicht klicken.

Man könnte den Eintrag in `resource_calendar_leaves` beim Löschen entfernen. Ich meine, das geht. Aber das löst leider nicht das Problem, dass dann ein falscher Eintrag stehen bleibt, wenn der Urlaubs-Antrag einfach geändert und wieder bestätigt wird :-(

**Comment by hbrunn** [comment]
ich schreibe gerade Tests fur die Falle und verdigado_employee, _officer, _manager, das wird uns diese lästigen regressions ersparen

File: verdigado_attendance/tests/test_holidays.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions
+
+from .hr_case import HrCase
+
+
+class TestHolidays(HrCase):
+    def test_standard_flow_past(self):
```
Comment (hbrunn) [inline verdigado_attendance/tests/test_holidays.py:10]: die Falle sind noch nicht vollständig, hier braucht es noch die ganzen Corner cases, Urlaub über Wochenenden, Feiertage, beides - hier bin ich auch etwas von Deinem Input abhängig

**Review by albig — APPROVED** [review]
